gc-design-system-variable-proposal.css

```diff
@@ -0,0 +1,167 @@
+
+/* Conceptually, the theming framework will have 2 sets of CSS variables for the repository, the first of which will determine the default/fallback values for the system, and the second set being the public-facing variables that can be used to "reskin" the system to meet third party needs. The first set of variables will have values that are determined by us, and reflect the Canada.ca theme. This means that anyone who installs the design system on their project can start building a site and it will default to the Canada.ca look and feel. If the user wants to reskin the system, they simply have to set their own variables (that we provide in a list format). */ 
+
+/* For example: */
+:root {
+    /* Canada.ca value, acts as a default/fallback. All --gc variables should have the value set to match the Canada.ca styleguide. These are not publicly available as a theming tool, they are only edited by us. */
+    --gc-font-serif: "Times New Roman", serif;
+    /* Public theme variable, if this variable is not specified by the user, the default/fallback is used. --gd variables are set by the user if they want to theme the system so that it no longer looks like the Canada.ca styleguide. */
+    --gd-font-serif: "Noto Serif", serif;
+}
+
+/* Using these variables in the design system CSS looks like this: */
+p {
+    /* GD Font Serif is used IF IT HAS BEEN DEFINED, otherwise, our fallback is used. */
+    font-family: var(--gd-font-serif, var(--gc-font-serif));
+}
+
+/* In practice, we'd have a list of all default variables, and then a separate list of public theme variables. The variables that we should define system wide depend on how much control we want/are able to relinquish to the user. I've proposed a list below based on my implementation of Clone, however, some of them might not be feasible for accessibility or tech reasons. Quite a few of the granular controls depend on things like calc() being present. */
+
+/* Core Layout Variables ==================================================== */
+/* These variables allow us to customize the value of default widths of page containers, how whitespace (i.e. margins and padding) is calculated, how whitespace responds to media queries, and how flexible the grid system is. */
+
+/* Container Values */
+/* These variables allow us to a) set a standard set of widths for pages, and b) allows the user of the system to change these values if the widths we've set don't suit their needs. */
+:root {
+    /* Small: used for containing text and ensuring readability of long swaths of text. */
+    --gc-layout-small: 39rem;
+    /* Medium: used for general UI. */
+    --gc-layout-medium: 60rem;
+    /* Large: used for large grids, or more complex UI that needs horizontal space. */
+    --gc-layout-large: 80rem;
```

Comment:
We migth want to get away from fixed units for font-size, for accessibility purposes. Fixed units doesn't scale well when zooming and resizing text.
https://www.24a11y.com/2019/pixels-vs-relative-units-in-css-why-its-still-a-big-deal/
https://usability.yale.edu/web-accessibility/articles/zoom-resizing-text

Reply:
According to the link you provided, `ch` seems more appropriate for count and seems not very precise for sizing, since it's an approximate measurement of a glyph
>This measurement is an approximation (and in monospace fonts, an exact measure) of a single narrow glyph’s advance measure, thus allowing measurements based on an expected glyph count.

Reply:
Let's try it out